File: 0014-Longest-Common-Prefix/lc0014.py
# Python solution.

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        for i in range(len(strs[0])):
            if any(i == len(strv) or strs[0][i] != strv[i] for strv in strs[1:]):
                return strs[0][:i]
        return strs[0]

File: 0014-Longest-Common-Prefix/lc0014.rs
// Rust solution.

impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        if strs.is_empty() { return "".to_string(); }
        let strvec = strs.iter().map(|s| s.as_bytes()).collect::<Vec<_>>();
        for (k, &c) in strvec[0].iter().enumerate() {
            for idx in 1..strs.len() {
                if k == strs[idx].len() || c != strvec[idx][k] {
                    return String::from(&strs[0][0..k]);
                }
            }
        }
        return String::from(&strs[0][..]);
    }
}
